### ISSUE TITLE
use-package ess no longer works

### ISSUE THREAD

**Issue by jabranham** [emacs-ess/ESS#827]
Opening this issue so we don't forget this breaking change before release: https://github.com/emacs-ess/ESS/commit/7df0059d88b55a21698963c14613a772dce8f0e0#commitcomment-31820472

**Comment by jabranham** [comment]
@vspinu @lionel- How shall we fix this one? I think the easiest/best way is to revert that commit so that ess-mode is defined in ess.el. I know some people find that annoying but otherwise we're introducing this breaking change for little reason. WDYT?

Another way is to autoload all our keymaps, which seems ugly to me. It also doesn't help users who don't use the autoloads.

**Comment by lionel-** [comment]
I'm not particularly worried about this having worked for only one version. There are probably few people who picked up this new way of configuring ess.

I think it would be fine to autoload the ESS and iESS maps from ess.el if that fixes it. If that's about the R mode, I think the `ess` feature should be independent from it, as ESSR should be a separate package.

**Comment by vspinu** [comment]
Yes. Let's autoload the two maps as @lionel- suggested, either in ess.el or directly. Preferably the latter if that fixes use package issues. `autoload` has 'keymap argument. 

Reverting is not an option as it doesn't fix derived maps like `ess-r-mode-map` for which users would still need `(use-package ess-r-mode  ...)`.
